lib/src/box.dart

```diff
@@ -163,75 +121,73 @@ class Box<T> {
 
         // ignore: omit_local_variable_types
         Pointer<Uint8> dataPtr = dataPtrPtr.value.cast<Uint8>();
-        final size = sizePtr.value;
-        return _fbManager.unmarshal(dataPtr, size);
+        return _entity.objectFromFB(dataPtr.asTypedList(sizePtr.value));
       });
     } finally {
       free(dataPtrPtr);
       free(sizePtr);
     }
   }
 
-  List<R> _getMany<R>(
-      List<R> Function(Pointer<OBX_bytes_array>) unmarshalArrayFn,
-      R Function(Pointer<Uint8>, int) unmarshalSingleFn,
-      Pointer<OBX_bytes_array> Function() cGetArray,
-      void Function(DataVisitor) cVisit) {
-    return _store.runInTransaction(TxMode.Read, () {
-      if (_supportsBytesArrays) {
-        final bytesArray = cGetArray();
-        try {
-          return unmarshalArrayFn(bytesArray);
-        } finally {
-          bindings.obx_bytes_array_free(bytesArray);
-        }
-      } else {
-        final results = <R>[];
-        final visitor = DataVisitor((Pointer<Uint8> dataPtr, int length) {
-          results.add(unmarshalSingleFn(dataPtr, length));
-          return true;
-        });
-
-        try {
-          cVisit(visitor);
-        } finally {
-          visitor.close();
-        }
-        return results;
-      }
-    });
-  }
-
   /// Returns a list of [ids.length] Objects of type T, each corresponding to the location of its ID in [ids].
   /// Non-existent IDs become null.
   List<T /*?*/ > getMany(List<int> ids) {
     if (ids.isEmpty) return [];
-
-    return executeWithIdArray(
-        ids,
-        (ptr) => _getMany(
-            _fbManager.unmarshalArrayWithMissing,
-            _fbManager.unmarshalWithMissing,
-            () => checkObxPtr(bindings.obx_box_get_many(_cBox, ptr),
-                'failed to get many objects from box'),
-            (DataVisitor visitor) => checkObx(bindings.obx_box_visit_many(
-                _cBox, ptr, visitor.fn, visitor.userData))));
+    return _store.runInTransactionWithPtr(TxMode.Read, (Pointer<OBX_txn> txn) {
+      final cCursor = checkObxPtr(bindings.obx_cursor(txn, _entity.model.id.id),
+          'failed to create cursor');
+      final dataPtrPtr = allocate<Pointer<Void>>();
+      final sizePtr = allocate<IntPtr>();
+      try {
+        final result = List<T>(ids.length);
+        for (var i = 0; i < ids.length; i++) {
+          final code =
+              bindings.obx_cursor_get(cCursor, ids[i], dataPtrPtr, sizePtr);
+          if (code == OBX_NOT_FOUND) {
+            result[i] = null;
+          } else {
+            checkObx(code);
+            result[i] = _entity.objectFromFB(
+                dataPtrPtr.value.cast<Uint8>().asTypedList(sizePtr.value));
+          }
+        }
+        return result;
+      } finally {
+        free(dataPtrPtr);
+        free(sizePtr);
+        checkObx(bindings.obx_cursor_close(cCursor));
+      }
+    });
   }
 
   /// Returns all stored objects in this Box.
   List<T> getAll() {
```

Comment:
Maybe document that `getAll()` and maybe `getMany()` now return fixed-size lists (as the default is growable, no?)?

E.g. imagine a use case like: get and then modify some objects and add some new objects, then calling putMany. That wouldn't work now.

Reply:
getAll actually returns a growable list (we don't know the size beforehand). 

However, you've inspired me to change getMany(), adding a `growableResult=false` named optional param, allowing user to pick whether they need the (slower) growable list

Reply:
`getAll()` is fine, not sure what I was seeing.

`getMany()` growable change: https://github.com/objectbox/objectbox-dart/commit/8a551e49c44e0192ca59abd39d148bb8274a1d80 LGTM!